lib/rspec/mocks/error_generator.rb

```diff
@@ -142,7 +142,7 @@ def expected_part_of_expectation_error(expected_received_count, expectation_coun
       def actual_method_call_args_description(count, args)
         method_call_args_description(args) ||
           if count > 0 && args.length > 0
-            " with arguments: #{args.inspect.gsub(/\A\[(.+)\]\z/, '(\1)')}"
+            " with arguments: #{format_args(*args)}"
           else
             ""
           end
```

Comment:
This method now looks almost identical to `expected_method_call_args_description`.  Can they be unified some how?